src/components/main-view/main-view.jsx

```diff
@@ -108,26 +109,37 @@ console.log(error);
           </Col>
           }} />
     
-          <Route path="movies/:movieId" render={({ match, history }) => {
+          <Route path="/movies/:movieId" render={({ match, history }) => {
+          if (!user) return <Col>
+          <LoginView onLoggedIn={user => this.onLoggedIn(user)} />;
+          </Col>
+
           return <Col md={8}>
           <MovieView movie={movies.find(m => m._id === match.params.movieId)} 
           onBackClick={() => history.goBack()} />
           </Col>
           }} />
 
           <Route path="/directors/:name" render={({ match, history }) => {
-          if (movies.length === 0) return <div className ="main-view" />;
+          
+          if (!user) return <Col>
+          <LoginView onLoggedIn={user => this.onLoggedIn(user)} />;
+          </Col>
+
           return <Col md={8}>
           <DirectorView director={movies.find(m => m.Director.Name === 
             match.params.Name).Director} onBackClick={() => history.goBack()} />
           </Col>
           }} />
 
           <Route path="/genres/:name" render={({ match, history }) => {
-          if (movies.length === 0) return <div className ="main-view" />;
```

Comment:
Similar to my previous comment, the number of movies shouldn't affect how this view is rendered.